UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/SalesCatalogueService.cs

```diff
@@ -0,0 +1,110 @@
+            return response; 
+        }
+
+        public async Task<SalesCatalogueResponse> PostProductIdentifiersAsync(List<string> ProductIdentifiers)
+        {
+            logger.LogInformation(EventIds.SCSPostProductIdentifiersRequestStart.ToEventId(), $"Post sales catalogue service for ProductIdentifiers Started");
+
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(salesCatalogueConfig.Value.ResourceId);
+            var uri = $"/{salesCatalogueConfig.Value.Version}/productData/{salesCatalogueConfig.Value.ProductType}/products/productIdentifiers";
+
+            string payloadJson = JsonConvert.SerializeObject(ProductIdentifiers);
+
+            var httpResponse = await salesCatalogueClient.CallSalesCatalogueServiceApi(HttpMethod.Post, payloadJson, accessToken, uri);
+
+            SalesCatalogueResponse response = await CreateSalesCatalogueServiceResponse(httpResponse);
+
+            logger.LogInformation(EventIds.SCSPostProductIdentifiersRequestCompleted.ToEventId(), $"Post sales catalogue service for ProductIdentifiers completed");
+            return response;
+        }
+
+        public async Task<SalesCatalogueResponse> PostProductVersionsAsync(List<ProductVersionRequest> productVersions)
+        {
+            logger.LogInformation(EventIds.SCSPostProductVersionsRequestStart.ToEventId(), $"Post sales catalouge service for ProductVersions started");
+
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(salesCatalogueConfig.Value.ResourceId);
+            var uri = $"/{salesCatalogueConfig.Value.Version}/productData/{salesCatalogueConfig.Value.ProductType}/products/productVersions";
+
+            string payloadJson = JsonConvert.SerializeObject(productVersions);
+
+            var httpResponse = await salesCatalogueClient.CallSalesCatalogueServiceApi(HttpMethod.Post, payloadJson, accessToken, uri);
+
+            SalesCatalogueResponse response = await CreateSalesCatalogueServiceResponse(httpResponse);
+            
+            logger.LogInformation(EventIds.SCSPostProductVersionsRequestCompleted.ToEventId(), $"Post sales catalogue service for ProductVersions completed");
+            return response;
+        }
+
+        private async Task<SalesCatalogueResponse> CreateSalesCatalogueServiceResponse(HttpResponseMessage httpResponse)
+        {
+            var response = new SalesCatalogueResponse();
+            var body = await httpResponse.Content.ReadAsStringAsync();
+            if (httpResponse.StatusCode != HttpStatusCode.OK && httpResponse.StatusCode != HttpStatusCode.NotModified)
+            {
+                logger.LogError(EventIds.SalesCatalogueNonOkResponse.ToEventId(), $"Sales catalougue service with uri {httpResponse.RequestMessage.RequestUri} and responded with {httpResponse.StatusCode} and message {body}");
```

Comment:
For all logs, can we ensure we use structured logging with parameters, rather than interpolated strings.

```suggestion
logger.LogError(EventIds.SalesCatalogueNonOkResponse.ToEventId(), "Sales catalougue service with uri {RequestUri} and responded with {StatusCode} and message {body}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode, body);
```

Reply:
did changes accordingly.